src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -551,6 +591,121 @@ public UpdateItemResult updateItem(UpdateItemRequest updateItemRequest) {
+            if (backupMetadata != null) {
+                return MtBackupAwsAdaptorKt.getBackupAdaptorSingleton().getDescribeBackupResult(backupMetadata);
+            } else {
+                throw new BackupNotFoundException("No backup with arn "
+                    + describeBackupRequest.getBackupArn() + "found");
+            }
+        } else {
+            throw new ContinuousBackupsUnavailableException("Backups can only be created by configuring a backup "
+                + "managed on an mt-dynamo table builder, see <insert link to backup guide>");
+        }
+    }
+
+    @Override
+    public CreateBackupResult createBackup(CreateBackupRequest createBackupRequest) {
+        if (backupManager.isPresent()) {
+            Preconditions.checkNotNull(createBackupRequest.getBackupName(), "Must pass backup name.");
+            Preconditions.checkArgument(createBackupRequest.getTableName() == null,
+                "Multitenant backups cannot backup individual tables, table name arguments are disallowed.");
+            backupManager.get().createBackup(createBackupRequest);
+
+            ExecutorService executorService = Executors.newFixedThreadPool(mtTables.keySet().size());
+
+            Set<String> snapshottedTables = Sets.newHashSet();
+            List<Future<SnapshotResult>> futures = Lists.newArrayList();
+            Set<String> origMtTables = ImmutableSet.copyOf(mtTables.keySet());
+            for (String tableName : origMtTables) {
+                String snapshottedTable = tableName + "-copy";
+                snapshottedTables.add(snapshottedTable);
+                mtTables.put(snapshottedTable, mtTables.get(tableName));
+                futures.add(executorService.submit(() ->
+                    backupManager.get()
+                        .getMtBackupTableSnapshotter()
+                        .snapshotTableToTarget(
+                            new SnapshotRequest(createBackupRequest.getBackupName(),
+                                tableName,
+                                snapshottedTable,
+                                getAmazonDynamoDb(),
+                                new ProvisionedThroughput(10L, 10L))
+                        )));
+            }
+            List<SnapshotResult> snapshotResults = Lists.newArrayList();
+            try {
+                for (Future<SnapshotResult> future : futures) {
+                    try {
+                        snapshotResults.add(future.get());
+                    } catch (InterruptedException | ExecutionException e) {
+                        throw new MtBackupException("Error snapshotting table", e);
+                    }
+                }
+            } finally {
+                executorService.shutdown();
+            }
+            try {
+                backupManager.get().getMtBackupTableSnapshotter();
+                for (SnapshotResult snapshotResult : snapshotResults) {
```

Comment:
Why not do this in parallel as well?

Reply:
makes sense, will include that in follow up PR

Reply:
addressed in https://github.com/salesforce/mt-dynamo/pull/525